openedx/core/lib/api/tests/test_parsers.py

```diff
@@ -0,0 +1,94 @@
+"""
+TestCases verifying proper behavior of custom DRF request parsers.
+"""
+
+from collections import namedtuple
+from io import BytesIO
+
+from rest_framework import exceptions
+from rest_framework.test import APITestCase, APIRequestFactory
+
+from openedx.core.lib.api import parsers
+
+
+class TestTypedFileUploadParser(APITestCase):
+    """
+    Tests that verify the behavior of TypedFileUploadParser
+    """
+    def setUp(self):
+        super(TestTypedFileUploadParser, self).setUp()
+        self.parser = parsers.TypedFileUploadParser()
+        self.request_factory = APIRequestFactory()
+        upload_media_types = {'image/png', 'image/jpeg', 'application/octet-stream'}
+        self.view = namedtuple('view', ('upload_media_types',))(upload_media_types)
+
+    def test_parse_supported_type(self):
+        """
+        Test that TypedFileUploadParser returns empty data and content stored in
+        files['file'].
+        """
+        request = self.request_factory.post(
+            '/',
+            content_type='image/png',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="file.PNG"',
+        )
+        context = {'view': self.view, 'request': request}
+        result = self.parser.parse(
+            stream=BytesIO('abcdefgh'), media_type='image/png', parser_context=context
+        )
+        self.assertEqual(result.data, {})
+        self.assertIn('file', result.files)
+        self.assertEqual(result.files['file'].read(), 'abcdefgh')
+
+    def test_parse_unsupported_type(self):
+        """
+        Test that TypedFileUploadParser raises an exception when parsing an
+        unsupported image format.
+        """
+        request = self.request_factory.post(
+            '/',
+            content_type='image/tiff',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="file.tiff"',
+        )
+        context = {'view': self.view, 'request': request}
+        with self.assertRaises(exceptions.UnsupportedMediaType):
+            self.parser.parse(
+                stream=BytesIO('abcdefgh'), media_type='image/tiff', parser_context=context
+            )
+
+    def test_parse_unconstrained_type(self):
+        """
+        Test that TypedFileUploader allows any extension for mimetypes without
+        specified extensions
+        """
+        request = self.request_factory.post(
+            '/',
+            content_type='application/octet-stream',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="VIRUS.EXE',
```

Comment:
Good.  I'm trying to communicate that the extension is unchecked if you allow types with unspecified extensions.